src/cdk/overlay/overlay.spec.ts

```diff
@@ -939,36 +961,35 @@ describe('Overlay', () => {
         .toContain('custom-panel-class');
 
       overlayRef.detach();
+      viewContainerFixture.changeDetectorRef.markForCheck();
       viewContainerFixture.detectChanges();
       expect(pane.classList).not.toContain('custom-panel-class', 'Expected class to be removed');
 
       overlayRef.attach(componentPortal);
+      viewContainerFixture.changeDetectorRef.markForCheck();
       viewContainerFixture.detectChanges();
       expect(pane.classList)
         .withContext('Expected class to be re-added')
         .toContain('custom-panel-class');
     });
 
-    it('should wait for the overlay to be detached before removing the panelClass', () => {
+    it('should wait for the overlay to be detached before removing the panelClass', async () => {
```

Comment:
This looks like the only test that needed to be changed. All of the tests seem to pass without any of the `markForCheck` additions in PR.